src/caret_analyze/architecture/architecture.py

```diff
@@ -297,3 +295,23 @@ def update_named_path(self, path_name: str, path_info: PathStructValue):
 
         self.remove_named_path(path_info.path_name)
         self.add_named_path(path_name, path_info)
+
+class Diff:
+    def __init__(
+        self,
+        arch1: Architecture,
+        arch2: Architecture
+    ):
+        pass
+
+    @staticmethod
+    def _diff_node_names(
+        arch1: Architecture,
+        arch2: Architecture
+    ) -> Tuple[Tuple[str, ...], Tuple[str, ...]]:
+        set_arch1 = set(arch1.node_names)
+        set_arch2 = set(arch2.node_names)
+        inter_arch1_arch2 = set_arch1 & set_arch2
```

Comment:
- [ ] It appears to be a redundant description.

```suggestion
        inter_arch1_arch2 = set(arch1.node_names) & set(arch2.node_names)
```

Reply:
After discussion, it is determined that the current implementation is acceptable.